src/components/Synthetics/TradeHistory/useDownloadAsCsv.tsx

```diff
@@ -0,0 +1,128 @@
+import { definedOrThrow } from "lib/guards";
+import { getSyntheticsGraphClient } from "lib/subgraph/clients";
+
+import { formatPositionMessage } from "./TradeHistoryRow/utils/position";
+import { RowDetails } from "./TradeHistoryRow/utils/shared";
+import { formatSwapMessage } from "./TradeHistoryRow/utils/swap";
+
+const GRAPHQL_MAX_SIZE = 10_000;
+
+export function useDownloadAsCsv({
+  marketAddresses,
+  forAllAccounts,
+  account,
+  fromTxTimestamp,
+  toTxTimestamp,
+  orderEventCombinations,
+  minCollateralUsd,
+}: {
+  marketAddresses: string[] | undefined;
+  forAllAccounts: boolean | undefined;
+  account: string | null | undefined;
+  fromTxTimestamp: number | undefined;
+  toTxTimestamp: number | undefined;
+  orderEventCombinations:
+    | {
+        eventName?: TradeActionType | undefined;
+        orderType?: OrderType | undefined;
+        isDepositOrWithdraw?: boolean | undefined;
+      }[]
+    | undefined;
+
+  minCollateralUsd?: BigNumber;
+}): [boolean, () => Promise<void>] {
+  const { chainId } = useChainId();
+  const marketsInfoData = useMarketsInfoData();
+  const tokensData = useTokensData();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleCsvDownload = useCallback(async () => {
+    try {
+      setIsLoading(true);
+
+      const client = getSyntheticsGraphClient(chainId);
+      definedOrThrow(client);
+
+      const tradeActions = await fetchTradeActions({
+        chainId,
+        pageIndex: 0,
+        pageSize: GRAPHQL_MAX_SIZE,
+        marketAddresses,
+        forAllAccounts,
+        account,
+        fromTxTimestamp,
+        toTxTimestamp,
+        orderEventCombinations,
+        marketsInfoData,
+        tokensData,
+      });
+
+      const fullFormattedData = tradeActions
+        .map((tradeAction) => {
+          const explorerUrl = getExplorerUrl(chainId) + `tx/${tradeAction.transaction.hash}`;
+
+          let rowDetails: RowDetails | null;
+
+          if (isSwapOrderType(tradeAction.orderType!)) {
+            rowDetails = formatSwapMessage(tradeAction as SwapTradeAction, marketsInfoData, false);
+          } else {
+            rowDetails = formatPositionMessage(tradeAction as PositionTradeAction, minCollateralUsd!, false);
+          }
+
+          return {
+            ...rowDetails,
+            explorerUrl,
+          };
+        })
+        .filter(Boolean);
+
+      const timezone = dateFnsFormat(new Date(), "z");
+
+      downloadAsCsv("trade-history", fullFormattedData, ["priceComment"], {
+        timestamp: t`Date` + ` (${timezone})`,
+        action: t`Action`,
+        size: t`Size`,
+        market: t`Market`,
+        fullMarket: t`Full market`,
+        marketPrice: t`Mark Price`,
+        acceptablePrice: t`Acceptable Price`,
+        executionPrice: t`Execution Price`,
+        triggerPrice: t`Trigger Price`,
+        priceImpact: t`Price Impact`,
+        explorerUrl: t`Transaction ID`,
+      });
+    } finally {
```

Comment:
let's also do some catch with `helperToast.error(...)` since it's a big request? 
We may ask @xhiroz for a better message here

Reply:
Synced with xhiros and updated error message